### PR TITLE
Refactor alerts based on backend reasons.

### PR DESCRIPTION
**Pull request by Maascha**

### CODE REVIEW

File: src/lang/lernfair/de.ts [hunk 1]
```diff
@@ -84,37 +84,36 @@ const lernfair = {
     schoolclass: '{{class}}. Klasse',
 
     reason: {
-        'subcourse-full': {
-            coursetext: 'Dieser Kurs ist leider grade ausgebucht.',
+        course: {
+            // from backend/graphql/student/fields.ts
```
Comment (Maascha) [inline src/lang/lernfair/de.ts:88]: Do we have something like "if this then that" to make sure the text stays in sync to the backend?

File: src/lang/lernfair/de.ts [hunk 2]
```diff
@@ -84,37 +84,36 @@ const lernfair = {
...
+            instructor: {
+                'not-instructor': 'Du bist noch nicht als Kursleiter:in registriert.',
+                'not-screened': 'Du bist noch nicht als Kursleter:in freigeschaltet. Bitte führe zuerst dein Kennenlerngespräch mit uns.',
```
Comment (Jonasdoubleyou) [inline src/lang/lernfair/de.ts:91]: typo

File: src/lang/lernfair/de.ts (hunk 1 above)
Comment (Jonasdoubleyou) [reply src/lang/lernfair/de.ts:88]: Nope, most semantic changes in the Backend will cause changes to the GraphQL Schema, which is part of the build process of the user app. The reason is a string though unfortunately. We could rework the Backend to use different subclasses of Reason, each with it's own enum for the reason field, then we could use mapped typed in Typescript here ... But that would need quite some rework ...

File: src/lang/lernfair/de.ts (hunk 2 above)
Comment (Maascha) [reply src/lang/lernfair/de.ts:91]: done

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**